pkg/propertyprovider/aks/controllers/pod.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers
```

Comment:
do we have integration tests for these controllers in the future PR?

Reply:
Those controllers are quite straightforward so at this moment I feel that they might not need individual integration tests (as the provider ones already cover all of their branches). We could always add some, though, if necessary